### PR TITLE
init

### PR DESCRIPTION
**Pull request by aaadryyy**
I'm not sure what to commit anymore but here it is 😄

### CODE REVIEW

**Comment by adipasquale** [comment]
screen shot 2018-12-27 at 10 02 50

that is much better ! In your previous PR there were 1200 files ! 

Still a few problems though : 

- you deleted the root `.gitignore` file. I don't think you wanted to do this. 
- you didn't commit the `client/package-lock.json` file (or `yarn.lock` if using yarn). this lock file should be committed.

The removal of `.gitignore` is pretty bad. Look what happens when I checkout your branch :
screen shot 2018-12-27 at 10 08 57

When you've committed and are ready to push, you should have no diff locally (meaning that running `git status` should tell you "nothing to commit, working tree clean".

Did you run `git diff --cached` before committing ? You should have seen something like that and notice the deleted `.gitignore` file : 
screen shot 2018-12-27 at 10 11 02

almost there :)

**Comment by adipasquale** [comment]
@aaadryyy here is a screencast of me committing on my blog repository : [http://recordit.co/1SFtMPL52G](http://recordit.co/1SFtMPL52G)

- I run git status like 10 times
- I start with `git add -p` to commit updates on text files chunk by chunk. In this case it's hard to read because the chunks are really big
- I then add new files or updated binary files one by one with `git add file`
- I make sure not to add files I don't want. for example the `.DS_Store` files you never want to commit. You can `rm .DS_Store` but they will re-appear as Mac Os always creates them. So I add them to the `.gitignore`, and then `git add .gitignore` so that it's taken into account. and poof, the `.DS_Store` files don't appear anymore in git status.

**Comment by aaadryyy** [comment]
Thanks for the tuto ! 🙏 
I did run `git diff --cached` the thing is I'm still not really sure about which files are finally necessary or not... 

I couldn't find the client/package-lock, I don't understand why... nor why do I still have those yarn things even though I never use no yarn command.

File: .gitignore [hunk 1]
```diff
@@ -17,3 +17,31 @@ api/tmp/*
...
+# dependencies
+client/node_modules
+client/.pnp
+client/.pnp.js
+
+# testing
+client//coverage
+
+# production
+client//build
```
Comment (adipasquale) [inline .gitignore]: extra slash

File: .gitignore [hunk 2]
```diff
@@ -17,3 +17,31 @@ api/tmp/*
...
+
+
+
+# dependencies
+client/node_modules
+client/.pnp
+client/.pnp.js
+
+# testing
+client//coverage
```
Comment (adipasquale) [inline .gitignore]: extra slash

File: .gitignore [hunk 3]
```diff
@@ -17,3 +17,31 @@ api/tmp/*
...
+client/.DS_Store
+client/.env.local
+client/.env.development.local
+client/.env.test.local
+client/.env.production.local
+
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+client/yarn.lock
```
Comment (adipasquale) [inline .gitignore:45]: I'm guessing these should all be prefixed with `client/` as we're going to run npm (or yarn) from inside the client folder.

File: api/package-lock.json [hunk 4]
```diff
@@ -0,0 +1,96 @@
+{
```
Comment (adipasquale) [inline api/package-lock.json]: this file should be inside `client/` not `api/`.

File: package-lock.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (adipasquale) [inline package-lock.json]: and this file should not exist at all :) you must have run `npm install` from the root folder, which is incorrect.

**Review by adipasquale — COMMENTED** [review]
almost there :)

**Comment by adipasquale** [comment]
if you don't have a `client/package-lock.json` it probably means that you have not run `npm install` from this `client` folder.

**Comment by aaadryyy** [comment]
is it ok ?

File: client/.gitignore [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Comment (adipasquale) [inline client/.gitignore]: haha sorry this file re-appeared somehow ! you can remove it. there needs to be only a single `.gitignore` at the root.

**Review by adipasquale — CHANGES_REQUESTED** [review]
really almost there now !

File: client/.gitignore (hunk 6 above)
Comment (aaadryyy) [reply client/.gitignore]: oopsy

**Review by adipasquale — APPROVED** [review]

**Comment by adipasquale** [comment]
ok thanks for your patience ! Last check before I merge : 
When I run `npm install` in the client folder, it works, but then I have a lot of diff in the `package-lock.json` file. Can you try re-running `npm install` in the client folder and tell me if you also get diff ? In this case, I think you should commit the `package-lock.json` updated version. 
It's important to make sure that we are working with the same versions of the dependencies (and the server)

**Comment by aaadryyy** [comment]
Ok, did run `npm install` in `client` but I have nothing when I run `git diff --cached`

**Comment by adipasquale** [comment]
`git diff --cached` is to see the diff that you have already "staged" using`git add`. You should simply use `git diff` here to check if running npm install updates your local `package-lock.json` file :

screen shot 2018-12-27 at 16 01 59

**Comment by aaadryyy** [comment]
Nothing with `git diff` either 🙈

**Comment by aaadryyy** [comment]
Nothing with `git diff` nor `git diff | head` either 🙈

**Pull request merged**